catalyst/callbacks/sklearn_model.py

```diff
@@ -0,0 +1,293 @@
+            optimizer = Adam(model.parameters(), lr=0.001)
+
+            # 3. criterion with triplets sampling
+            sampler_inbatch = data.HardTripletsSampler(norm_required=False)
+            criterion = nn.TripletMarginLossWithSampler(
+                margin=0.5,
+                sampler_inbatch=sampler_inbatch
+            )
+
+            # 4. training with catalyst Runner
+            class CustomRunner(dl.SupervisedRunner):
+                def handle_batch(self, batch) -> None:
+                    images, targets = batch["features"].float(), batch["targets"].long()
+                    features = self.model(images)
+                    self.batch = {
+                        "embeddings": features,
+                        "targets": targets,
+                    }
+
+            callbacks = [
+                dl.ControlFlowCallback(
+                    dl.CriterionCallback(
+                        input_key="embeddings",
+                        target_key="targets",
+                        metric_key="loss"),
+                    loaders="train",
+                ),
+                dl.SklearnModelCallback(
+                    feature_key="embeddings",
+                    target_key="targets",
+                    train_loader="train",
+                    valid_loader="valid",
+                    sklearn_classifier_fn="linear_model.LogisticRegression",
+                    predict_method="predict_proba",
+                    predict_key="sklearn_predict"
+                ),
+                dl.ControlFlowCallback(
+                    dl.AccuracyCallback(
+                        target_key="targets", input_key="sklearn_predict", topk_args=(1, 3)
+                    ),
+                    loaders="valid"
+                )
+            ]
+
+            runner = CustomRunner(input_key="features", output_key="embeddings")
+            runner.train(
+                model=model,
+                criterion=criterion,
+                optimizer=optimizer,
+                callbacks=callbacks,
+                loaders={"train": train_loader, "valid": valid_loader},
+                verbose=False,
+                logdir="./logs",
+                valid_loader="valid",
+                valid_metric="accuracy",
+                minimize_valid_metric=False,
+                num_epochs=100,
+            )
+
+    """
+
+    def __init__(
+        self,
+        feature_key: str,
+        target_key: str,
+        train_loader: str,
+        valid_loader: str,
+        sklearn_classifier_fn: Union[Callable, str],
+        predict_method: str = "predict",
+        predict_key: str = "sklearn_predict",
+        **classifier_kwargs,
+    ) -> None:
+        super().__init__(order=CallbackOrder.Metric)
+
+        if isinstance(sklearn_classifier_fn, str):
+            base, clf = sklearn_classifier_fn.split(".")
+            base = f"sklearn.{base}"
+            sklearn_classifier_fn = getattr(importlib.import_module(base), clf)

```

Comment:
let's make it with `hydra-slayer`

Reply:
Could you give me advice on how to do it?